File: FrontendAng/src/app/Core/Services/program.service.ts
import {HttpClient} from "@angular/common/http";
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {Program} from "../Models/program";


@Injectable({
  providedIn: 'root'
})
export class ProgramService {
  private baseUrl = 'http://localhost:8082/SpringMVC/api/programs';

  constructor(private http: HttpClient) { }

  getAllPrograms(): Observable<Program[]> {
    return this.http.get<Program[]>(`${this.baseUrl}`);
  }

  addProgram(program: Program, userId: number): Observable<Program> {
    return this.http.post<Program>(`${this.baseUrl}/${userId}`, program);
  }

  getProgramById(id: number): Observable<Program> {
    return this.http.get<Program>(`${this.baseUrl}/${id}`);
  }

  updateProgram(id: number, program: Program): Observable<Program> {
    return this.http.put<Program>(`${this.baseUrl}/${id}`, program);
  }


  deleteProgram(id: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: Backendspr/src/main/java/com/example/backendspr/Services/Interfaces/ProgramService.java
package com.example.backendspr.Services.Interfaces;
import com.example.backendspr.Models.Program;

import java.util.List;
import java.util.Optional;

public interface ProgramService {
    List<Program> getAllPrograms();

    Optional<Program> getProgramById(Long id);

    Program addProgram(Program program, Long userId);

    Program updateProgram(Long id,Program program);
    void deleteProgram(Long id);
}

